executor/projection.go

```diff
@@ -216,6 +225,9 @@ func (e *ProjectionExec) prepare(ctx context.Context) {
 
 // Close implements the Executor Close interface.
 func (e *ProjectionExec) Close() error {
+	if e.isUnparallelExec() {
```

Comment:
why we need this check?

Reply:
it paired for [Open](https://github.com/pingcap/tidb/pull/7076/files/56afa90414ebdb4caa97ceff1529993ceaf52e7b#diff-da03bd2739194a5e817f1b2a9685509bR77). It seems that alloc/dealloc child result chunk is required only when `isUnparallel==true`